source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -354,6 +369,15 @@ Http::FilterHeadersStatus OAuth2Filter::decodeHeaders(Http::RequestHeaderMap& he
   return Http::FilterHeadersStatus::StopAllIterationAndBuffer;
 }
 
+Http::FilterHeadersStatus OAuth2Filter::encodeHeaders(Http::ResponseHeaderMap& headers, bool) {
+  if (response_headers_to_add_) {
+    Http::HeaderMapImpl::copyFrom(headers, *response_headers_to_add_);
+    response_headers_to_add_.reset();
+  }
```

Comment:
Sorry. Could you add a comment on why we need this condition? Thank you.

Reply:
Because response_headers_to_add_ is not null only in the case of refreshing token. In other cases it is null